core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -331,7 +331,15 @@ private[spark] class MemoryStore(
     var unrollMemoryUsedByThisBlock = 0L
     // Underlying buffer for unrolling the block
     val redirectableStream = new RedirectableOutputStream
-    val bbos = new ChunkedByteBufferOutputStream(initialMemoryThreshold.toInt, allocator)
+    val chunkSize = if (initialMemoryThreshold > Int.MaxValue) {
+      logWarning(s"Initial memory threshold of ${Utils.bytesToString(initialMemoryThreshold)} " +
+        s"is too large to be set as chunk size. It is safe to be cap at Int.MaxValue for chunk " +
+        s"size instead.")
+      Int.MaxValue
+    } else {
+      initialMemoryThreshold.toInt
+    }
+    val bbos = new ChunkedByteBufferOutputStream(chunkSize, allocator)
```

Comment:
No, it's almost never a good idea, certainly not just to 'work around' bad config.
I think this is getting overly-complex. The net-net of a lot of discussion is that all we wanted to do was fix the actual bug, that a value > Int.MaxValue is not merely way too large, but read as the wrong number.